src/java.desktop/share/classes/javax/swing/plaf/basic/BasicComboBoxUI.java

```diff
@@ -1132,21 +1127,24 @@ public Accessible getAccessibleChild(JComponent c, int i) {
         // 0 = the popup
         // 1 = the editor
         switch ( i ) {
-        case 0:
-            if ( popup instanceof Accessible ) {
-                AccessibleContext ac = ((Accessible) popup).getAccessibleContext();
-                ac.setAccessibleParent(comboBox);
-                return(Accessible) popup;
-            }
-            break;
-        case 1:
-            if ( comboBox.isEditable()
-                 && (editor instanceof Accessible) ) {
-                AccessibleContext ac = ((Accessible) editor).getAccessibleContext();
-                ac.setAccessibleParent(comboBox);
-                return(Accessible) editor;
-            }
-            break;
+            case 0:
+                if ( popup instanceof Accessible ) {
+                    AccessibleContext ac = ((Accessible) popup).getAccessibleContext();
+                    ac.setAccessibleParent(comboBox);
+                    return (Accessible) popup;
+                }
+                break;
+            case 1:
+                if (comboBox.isEditable()
+                        && (editor instanceof Accessible)) {
+                    AccessibleContext ac = ((Accessible) editor).getAccessibleContext();
+                    ac.setAccessibleParent(comboBox);
+                    return (Accessible) editor;
+                } else if (!comboBox.isEditable()
+                        && (arrowButton instanceof Accessible)) {
+                    return (Accessible) arrowButton;
+                }
```

Comment:
How about to idiomatize the code a bit by using Java 14's enhanced `instanceof`?

Reply:
Yeah it can be done to use enhanced `instanceOf`.